joblib/test/test_numpy_pickle.py

```diff
@@ -378,14 +470,23 @@ def test_joblib_pickle_across_python_versions():
     class SubArray(np.ndarray):
 
         def __reduce__(self):
-            return (_load_sub_array, (np.asarray(self), ))
-
+            return _load_sub_array, (np.asarray(self), )
 
     def _load_sub_array(arr):
         d = SubArray(arr.shape)
         d[:] = arr
         return d
 
+    class ComplexTestObject:
+        """A complex object containing multiple np array memebers."""
+
+        def __init__(self):
+            """Initialize different types of arrays."""
+            self.array_float = np.arange(100, dtype='float64')
+            self.array_int = np.ones(100, dtype='int32')
+            self.array_obj = np.array(['a', 10, 20.0], dtype='object')
+
+
 @with_numpy
 def test_numpy_subclass():
```

Comment:
Could you also check the content of SubArray in this test (similar to what you did with matrix above)?